CICD.Tools.CatalogUpload/Program.cs

```diff
@@ -86,116 +103,115 @@ public static async Task<int> Main(string[] args)
 				IsRequired = false,
 			};
 
+			var catalogIdentifier = new Option<string>(
+			name: "--catalog-identifier",
+			description: "The GUID identifying the catalog item on https://catalog.dataminer.services/. When not provided, this should be present through a catalog.yml file as described here https://docs.dataminer.services/user-guide/Cloud_Platform/Catalog/Register_Catalog_Item.html#manifest-file.")
+			{
+				IsRequired = false,
+			};
+
+			var catalogDetailsYml = new Option<string>(
+			name: "--path-to-catalog-yml",
+			description: "Path to a yml file containing catalog details as described here https://docs.dataminer.services/user-guide/Cloud_Platform/Catalog/Register_Catalog_Item.html#manifest-file")
+			{
+				IsRequired = true,
+			};
+
+			var readme = new Option<string>(
+			name: "--path-to-readme",
+			description: "Path to a readme file written in markdown.")
+			{
+				IsRequired = true,
+			};
+
+			var images = new Option<string>(
+			name: "--path-to-images",
+			description: "Path to a folder with images used in the readme file.")
+			{
+				IsRequired = true,
+			};
+
+			// dataminer-catalog-upload
+			var rootCommand = new RootCommand("Uploads artifacts or their registration info to the artifact storage for the DataMiner Catalog and Cloud Connected Systems. (The default upload has no additional registration and no visibility on the catalog. Use the returned Artifact ID for deployment or download.)");
+			rootCommand.AddOption(pathToArtifactRequired); // No longer global due to onlyRegistration option
+			rootCommand.AddGlobalOption(dmCatalogToken);
+			rootCommand.AddGlobalOption(isDebug);
+
 			var withRegistrationCommand = new Command("with-registration", "Uploads artifacts to become visible in the Skyline DataMiner catalog (https://catalog.dataminer.services")
 			{
-				registrationIdentifier,
+				pathToArtifactOptional,
+				uriSourceCode,
 				overrideVersion,
 				branch,
 				committerMail,
-				releaseUri
+				releaseUri,
+				catalogIdentifier
 			};
 
-			rootCommand.SetHandler(Process, pathToArtifact, dmCatalogToken, isDebug);
-			withRegistrationCommand.SetHandler(ProcessWithRegistrationAsync, pathToArtifact, dmCatalogToken, isDebug, registrationIdentifier, overrideVersion, branch, committerMail, releaseUri);
+			var withOnlyRegistrationCommand = new Command("with-only-registration", "Uploads only the registration information for a Skyline DataMiner catalog (https://catalog.dataminer.services) item.")
```

Comment:
Confusing names. 'OnlyRegistration' is the registration of the item and 'Registration' is the registration of a version. Maybe look into renaming these.

Reply:
changed to update-catalog-details  here and in the readme. Left the sourcecode alone to avoid introducing new bugs.